source/Core/Membership/Memberships/MembershipBaseRepository.cs

```diff
@@ -0,0 +1,14 @@
+using Microsoft.EntityFrameworkCore;
+
+namespace NEBA.Membership.Memberships
+{
+    internal class Repository : Database.ReadOnlyRepository<Entities.MembershipBase>
+    {
+        private readonly DbSet<Entities.MembershipBase> _memberships;
+
+        protected internal Repository(MembershipsContext membershipContext) : base(membershipContext)

```

Comment:
this is not abstract so should be just internal
```suggestion
       internal Repository(MembershipsContext membershipContext) : base(membershipContext)

```